client/txn.go

```diff
@@ -77,30 +85,192 @@ type txn struct {
+	return err
+}
+
+// Inc increments the integer value at key. If the key does not exist it will
+// be created with an initial value of 0 which will then be incremented. If the
+// key exists but was set using Put or CPut an error will be returned.
+//
+// The returned Result will contain a single row and Result.Err will indicate
+// success or failure.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (tx *Tx) Inc(key interface{}, value int64) (KeyValue, error) {
+	return runOneRow(tx, tx.B.Inc(key, value))
+}
+
+// Scan retrieves the rows between begin (inclusive) and end (exclusive).
+//
+// The returned []KeyValue will contain up to maxRows elements.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (tx *Tx) Scan(begin, end interface{}, maxRows int64) ([]KeyValue, error) {
+	r, err := runOneResult(tx, tx.B.Scan(begin, end, maxRows))
+	return r.Rows, err
+}
+
+// Del deletes one or more keys.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (tx *Tx) Del(keys ...interface{}) error {
+	_, err := runOneResult(tx, tx.B.Del(keys...))
+	return err
+}
+
+// DelRange deletes the rows between begin (inclusive) and end (exclusive).
+//
+// The returned Result will contain 0 rows and Result.Err will indicate success
+// or failure.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (tx *Tx) DelRange(begin, end interface{}) error {
+	_, err := runOneResult(tx, tx.B.DelRange(begin, end))
+	return err
+}
+
+// Run executes the operations queued up within a batch. Before executing any
+// of the operations the batch is first checked to see if there were any errors
+// during its construction (e.g. failure to marshal a proto message).
+//
+// The operations within a batch are run in parallel and the order is
+// non-deterministic. It is an unspecified behavior to modify and retrieve the
+// same key within a batch.
+//
+// Upon completion, Batch.Results will contain the results for each
+// operation. The order of the results matches the order the operations were
+// added to the batch.
+func (tx *Tx) Run(b *Batch) error {
+	if err := b.prepare(); err != nil {
+		return err
+	}
+	if err := tx.send(b.calls...); err != nil {
+		return err
+	}
+	return b.fillResults()
+}
+
+// Commit executes the operations queued up within a batch and commits the
+// transaction. Explicitly committing a transaction is optional, but more
+// efficient than relying on the implicit commit performed when the transaction
+// function returns without error.
+func (tx *Tx) Commit(b *Batch) error {
+	args := &proto.EndTransactionRequest{Commit: true}
+	reply := &proto.EndTransactionResponse{}
+	b.calls = append(b.calls, Call{Args: args, Reply: reply})
+	b.initResult(1, 0, nil)
+	return tx.Run(b)
+}
+
+func (tx *Tx) init(db *DB) {
```

Comment:
Yeah, I did this originally when I didn't want `txn` to be allocated separately from `Tx`, but now that they are merged adding a `newTx` function seems better.